docs/services/deployment/cdk.md

```diff
@@ -0,0 +1,132 @@
+on the SDK and programming language, you may need an additional build process such as Gradle, to bundle your business
+logic in a Lambda-deployable artifact. 
+
+For example, to deploy a Java or Kotlin service handler to Lambda, you might use the following CDK construct:
+
+```typescript
+import * as lambda from "aws-cdk-lib/aws-lambda";
+
+const service: lambda.Function = new lambda.Function(scope, "RestateService", {
+  runtime: lambda.Runtime.JAVA_21,
+  architecture: lambda.Architecture.ARM_64,
+  code: lambda.Code.fromAsset(".../lambda-all.jar"), // use the path of your JAR file artifact
+  handler: "com.example.service.package.LambdaHandler", // use the fully qualified name of your handler class
+  timeout: cdk.Duration.seconds(10),
+});
+```
+
+For TypeScript/JavaScript services, you can use the [`NodejsFunction`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda_nodejs-readme.html)
+construct to transpile and bundle all the required dependencies.
+
+```typescript
+import * as lambda from "aws-cdk-lib/aws-lambda";
+import { NodejsFunction } from "aws-cdk-lib/aws-lambda-nodejs";
+
+const service = new NodejsFunction(scope, "RestateService", {
+  runtime: lambda.Runtime.NODEJS_18_X,
+  architecture: lambda.Architecture.ARM_64,
+  entry: "handler", // use the path to your service handler
+});
+```
+
+These definitions are no different to how you might deploy any other Lambda handler.
+
+### Register Lambda-based services with Restate
+
+To enable our Lambda-deployed handlers to receive Restate requests, we need to register them with a Restate instance.
+We can model existing Restate Cloud instance using the `RestateCloudEndpoint` construct:
+
+```typescript
+import * as restate from "@restatedev/restate-cdk";
+
+const restateInstance = new restate.RestateCloudEndpoint(scope, "RestateCloud", {
+  clusterId: "restate-cluster-id",
+  authTokenSecretArn: restateAuthToken.secretFullArn,
+});
+```
+
+Automatically register service handlers with Restate on deployment:
+
+```typescript
+import * as restate from "@restatedev/restate-cdk";
+
+const handlers = new restate.LambdaServiceRegistry(scope, "RestateServiceRegistry", {
+  serviceHandlers: {
+    "namespace.Service": service,
+  },
+  restate: restateInstance,
+});
+handlers.register({
+  metaEndpoint: restateInstance.metaEndpoint,
+  invokerRoleArn: restateInstance.invokerRole.roleArn,
+  authTokenSecretArn: restateInstance.authToken.secretArn,
+});
+```
+
+### Deploy a single-node development Restate instance
+
+Restate is [very easy to deploy](https://docs.restate.dev/restate/deployment), and we provide a CDK construct to take
+advantage of this to deploy a standalone self-hosted instance running on Amazon EC2. The `SingleNodeRestateInstance`
+construct deploys a single-node server and creates a new role that can be assumed by the Restate managed service with
```

Comment:
In the previous sentence, it says `self-hosted`, and in this one it says `Restate managed service`. This is a bit confusing.

Reply:
My mistake – this is poorly written. The reality is that there is an IAM service role for Restate use in either case, whether it's ours (managed/cloud service) or self-hosted.